### ISSUE TITLE
Typescript JSX errors not showing up

### ISSUE THREAD

**Issue by lipedeitos** [Microsoft/TypeScript#10595]
After spending several hours researching in various sources like stackoverflow unsuccessfully I decided to make a post here.
I am building an application with Typescript/React/VSCode, using JSX syntax I am having difficulty viewing development and compile errors. But when i use JS instead of JSX the VSCode intellisense works perfectly and prompt compile errors show up as expected.

**TypeScript Version:**  1.8.10

**TextField Code**

```ts
import * as React from 'react';
//edited here
interface ITextFieldProps {
    id: number,
    label: string
}

export default class TextField extends React.Component<ITextFieldProps, any> {
    render() { ... }
}
```

**Parent Component rendering TextField in JSX**
Event if "id" and "label" are required the VSCode and the Prompt don't show any error.
```ts
import * as React from 'react';
import TextField from 'textfield';

export default class Parent extends React.Component<any, any> {
    render() {
        return (
            <TextField />
        );
    }
}
```

**Parent Component rendering TextField in JS**
Here VSCode and the typescript compiler show errors saying that the component need the props
```ts
render() {
    return React.createElement(TextField, {});
}
```

**Comment by RyanCavanaugh** [comment]
I can't find the other bug, but @yuit (or maybe someone else?) fixed this at some point since 1.8. Verified fixed in 2.0 build.

**Comment by niba** [comment]
@RyanCavanaugh I have the same problem as @lipedeitos. Intelisense/compiler doesn't work for me at all in JSX. It worked in 1.8 version but doesn't work now (2.0@rc and 2.0@next)

**Comment by RyanCavanaugh** [comment]
@niba can you log a new issue outlining your setup? Thanks!

**Comment by lipedeitos** [comment]
Just so u guys know, i'm using "typescript": "^2.0.0" (beta) globally and in my project as a devDep.
Those errors are still happening and i found another similar problem using the same "ITextFieldProps" example above but this time using JS syntax too.
Writing React.createElement without second parameter even the JS syntax don't throw any error at the VSCode and compilation.

```
render() {
    return React.createElement(TextField);
}
```

**Comment by mhegazy** [comment]
by default VSCode comes in with TS 1.8, to get TS 2.0.2 you will need to set `typescript.sdk`, see https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Nightly%20Builds.md#visual-studio-code for more details.

**Comment by niba** [comment]
@RyanCavanaugh here you have #10658
